acquire/collector.py

```diff
@@ -362,9 +366,17 @@ def collect_symlink(
             outpath = self.create_outpath(path)
             entry = path.get()
 
-            self.output.write_bytes(outpath, b"", 0, entry)
-            self.report.add_symlink_collected(module_name, path)
+            if follow:
+                resolved_path = path.resolve()
+                if resolved_path.is_file():
+                    self.collect_file(resolved_path, size, module_name=module_name, volatile=volatile)
```

Comment:
You didn't end up doing this logic in `collect_path`? It's kinda weird now, because it generally doesn't make sense to pass a size to collect a symlink, because it doesn't have a size.

Reply:
I find that by moving the logic to the `collect_symlink` it made `collect_path` cleaner. You are right about the `size`. After some testing the size didn't end up being needed. So the parameter can be removed. This should make  `collect_symlink` make more sense.